python_ta/checkers/top_level_code_checker.py

```diff
@@ -53,12 +53,15 @@ def _is_allowed_assignment(statement: nodes.NodeNG) -> bool:
     """
     Return whether or not <statement> is a constant assignment or type alias assignment.
     """
-    if not isinstance(statement, nodes.Assign):
+    if not isinstance(statement, nodes.Assign) and not isinstance(statement, nodes.AnnAssign):
         return False
 
     names = []
-    for target in statement.targets:
-        names.extend(node.name for node in target.nodes_of_class(nodes.AssignName, nodes.Name))
+    if isinstance(statement, nodes.Assign):
+        for target in statement.targets:
+            names.extend(node.name for node in target.nodes_of_class(nodes.AssignName, nodes.Name))
+    else:
+        names.append(statement.target.name)
```

Comment:
While you have correctly identified `statement.target` as being singular (as opposed to the `.targets` above), you should be careful not to assume a particular class. Check out the astroid documentation for `nodes.AnnAssign`: it shows different possible types for the `target` attribute.

So, I still recommend using `nodes_of_class` here.